From 6bfa472e5ddb90e8b08972fbc19e9d156675e2f4 Mon Sep 17 00:00:00 2001
From: Jaroslav Mracek <jmracek@redhat.com>
Date: Wed, 18 Oct 2017 09:26:42 +0200
Subject: [PATCH 4/4] fixup! Not require spaces in provides (RhBug:1480176)

---
 libdnf/hy-iutil.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/libdnf/hy-iutil.c b/libdnf/hy-iutil.c
index 18bcad5..63fc02c 100644
--- a/libdnf/hy-iutil.c
+++ b/libdnf/hy-iutil.c
@@ -694,13 +694,13 @@ parse_reldep_str(const char *reldep_str, char **name, char **evr,
     regex_t reg;
     const char *regex =
         "^([^ \t\r\n\v\f<=>!]*)\\s*(<=|>=|!=|<|>|=)?\\s*(.*)$";
-    regmatch_t matches[6];
+    regmatch_t matches[4];
     *cmp_type = 0;
     int ret = 0;
 
     regcomp(&reg, regex, REG_EXTENDED);
 
-    if(regexec(&reg, reldep_str, 6, matches, 0) == 0) {
+    if(regexec(&reg, reldep_str, 4, matches, 0) == 0) {
         if (copy_str_from_subexpr(name, reldep_str, matches, 1) == -1)
             ret = -1;
         // without comparator and evr
-- 
2.13.5

