diff -up libdnf-0.16.1/docs/hawkey/CMakeLists.txt.2~ libdnf-0.16.1/docs/hawkey/CMakeLists.txt
--- libdnf-0.16.1/docs/hawkey/CMakeLists.txt.2~	2018-07-29 01:19:27.772333839 +0200
+++ libdnf-0.16.1/docs/hawkey/CMakeLists.txt	2018-07-29 01:20:23.718418029 +0200
@@ -2,9 +2,9 @@
 # tell sphinx-build to do them both in one go:
 
 if (${PYTHON_VERSION_MAJOR} STREQUAL "2")
-    SET(SPHINX_BUILD_NAME "sphinx-build")
+    SET(SPHINX_BUILD_NAME "sphinx-build-2")
 else()
-    SET(SPHINX_BUILD_NAME "sphinx-build-3")
+    SET(SPHINX_BUILD_NAME "sphinx-build")
 endif()
 
 ADD_CUSTOM_TARGET (doc-html
