diff -up libdnf-0.15.1/docs/hawkey/CMakeLists.txt.1~ libdnf-0.15.1/docs/hawkey/CMakeLists.txt
--- libdnf-0.15.1/docs/hawkey/CMakeLists.txt.1~	2018-06-29 01:01:56.851104418 +0200
+++ libdnf-0.15.1/docs/hawkey/CMakeLists.txt	2018-06-29 01:07:57.284117858 +0200
@@ -2,9 +2,9 @@
 # tell sphinx-build to do them both in one go:
 
 if (${PYTHON_DESIRED} STREQUAL "2")
-    SET(SPHINX_BUILD_NAME "sphinx-build")
+    SET(SPHINX_BUILD_NAME "sphinx-build-2")
 else()
-    SET(SPHINX_BUILD_NAME "sphinx-build-3")
+    SET(SPHINX_BUILD_NAME "sphinx-build")
 endif()
 
 ADD_CUSTOM_TARGET (doc-html
