From 6b2ef7ad4ed730dc74b97ffd7ef0ec4b66639226 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Bernhard=20Rosenkr=C3=A4nzer?= <bero@lindev.ch>
Date: Sun, 1 Apr 2018 10:30:03 +0200
Subject: [PATCH] Add armv8 (aarch64 in 32-bit mode) to arm/armhfp arches
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Signed-off-by: Bernhard Rosenkränzer <bero@lindev.ch>
---
 libdnf/dnf-context.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/libdnf/dnf-context.c b/libdnf/dnf-context.c
index 10d8f992..ca4e6c53 100644
--- a/libdnf/dnf-context.c
+++ b/libdnf/dnf-context.c
@@ -67,8 +67,8 @@ static const struct {
     { "alpha",      { "alpha", "alphaev4", "alphaev45", "alphaev5",
                       "alphaev56", "alphaev6", "alphaev67",
                       "alphaev68", "alphaev7", "alphapca56", NULL } },
-    { "arm",        { "armv5tejl", "armv5tel", "armv5tl", "armv6l", "armv7l", NULL } },
-    { "armhfp",     { "armv7hl", "armv7hnl", NULL } },
+    { "arm",        { "armv5tejl", "armv5tel", "armv5tl", "armv6l", "armv7l", "armv8l", NULL } },
+    { "armhfp",     { "armv7hl", "armv7hnl", "armv8hl", "armv8hnl", "armv8hcnl", NULL } },
     { "i386",       { "i386", "athlon", "geode", "i386",
                       "i486", "i586", "i686", NULL } },
     { "ia64",       { "ia64", NULL } },
