diff -Naur libdnf-0.67.0/libdnf/dnf-context.cpp libdnf-0.67.0.tpg/libdnf/dnf-context.cpp
--- libdnf-0.67.0/libdnf/dnf-context.cpp	2022-04-27 06:25:21.000000000 +0000
+++ libdnf-0.67.0.tpg/libdnf/dnf-context.cpp	2022-05-25 20:06:37.309080046 +0000
@@ -98,7 +98,7 @@
     { "arm",        { "armv5tejl", "armv5tel", "armv5tl", "armv6l", "armv7l", "armv8l", NULL } },
     { "armhfp",     { "armv6hl", "armv7hl", "armv7hnl", "armv8hl", "armv8hnl", "armv8hcnl", NULL } },
     { "i386",       { "i386", "athlon", "geode", "i386",
-                      "i486", "i586", "i686", NULL } },
+                      "i486", "i586", "i686", "znver1_32", NULL } },
     { "ia64",       { "ia64", NULL } },
     { "mips",       { "mips", NULL } },
     { "mipsel",     { "mipsel", NULL } },
@@ -118,7 +118,7 @@
     { "sh4",        { "sh4", "sh4a", NULL } },
     { "sparc",      { "sparc", "sparc64", "sparc64v", "sparcv8",
                       "sparcv9", "sparcv9v", NULL } },
-    { "x86_64",     { "x86_64", "amd64", "ia32e", NULL } },
+    { "x86_64",     { "x86_64", "amd64", "ia32e", "znver1", NULL } },
     { "loongarch64",     { "loongarch64", NULL } },
     { NULL,         { NULL } }
 };
diff -Naur libdnf-0.67.0/libdnf/hy-util.cpp libdnf-0.67.0.tpg/libdnf/hy-util.cpp
--- libdnf-0.67.0/libdnf/hy-util.cpp	2022-04-27 06:25:21.000000000 +0000
+++ libdnf-0.67.0.tpg/libdnf/hy-util.cpp	2022-05-25 20:08:45.818367597 +0000
@@ -60,6 +60,35 @@
 #define HWCAP_ARM_NEON	(1<<12)
 #endif
 
+#if defined(__x86_64__) || defined(__i386__)
+
+static inline void cpuid(uint32_t op, uint32_t *eax, uint32_t *ebx, uint32_t *ecx, uint32_t *edx)
+{
+    asm volatile (
+       "cpuid\n"
+    : "=a" (*eax), "=b" (*ebx), "=c" (*ecx), "=d" (*edx)
+    : "a" (op));
+}
+
+static bool is_ryzen() {
+    uint32_t eax, ebx, ecx, edx;
+    char vendor[13];
+    int family;
+    vendor[12]=0;
+    cpuid(0, &eax, &ebx, &ecx, &edx);
+    memcpy(vendor, &ebx, sizeof(ebx));
+    memcpy(vendor+4, &edx, sizeof(edx));
+    memcpy(vendor+8, &ecx, sizeof(ecx));
+    if (strncmp(vendor, "AuthenticAMD", 12))
+        return false;
+    cpuid(1, &eax, &ebx, &ecx, &edx);
+    family = (eax>>8)&0xf;
+    if(family == 0xf)
+        family += (eax>>20)&0x7f;
+    return family >= 0x17;
+}
+#endif
+
 const char *
 hy_chksum_name(int chksum_type)
 {
@@ -140,6 +169,14 @@
     else if (!strcmp(un.machine, "mips64"))
         strcpy(un.machine, "mips64el");
 #endif
+#if defined(__x86_64__) || defined(__i386__)
+    if (is_ryzen()) {
+        if (!strcmp(un.machine, "x86_64"))
+            strcpy(un.machine, "znver1");
+       else
+            strcpy(un.machine, "znver1_32");
+    }
+#endif
     *arch = g_strdup(un.machine);
     return 0;
 }
