diff -up libdnf-0.28.0/libdnf/CMakeLists.txt.2~ libdnf-0.28.0/libdnf/CMakeLists.txt
--- libdnf-0.28.0/libdnf/CMakeLists.txt.2~	2019-03-12 12:34:15.705189946 +0100
+++ libdnf-0.28.0/libdnf/CMakeLists.txt	2019-03-12 12:34:35.166293161 +0100
@@ -73,6 +73,7 @@ target_link_libraries(libdnf
     ${LIBMODULEMD_LIBRARIES}
     ${SMARTCOLS_LIBRARIES}
     ${GPGME_VANILLA_LIBRARIES}
+    dl
 )
 
 if(ENABLE_RHSM_SUPPORT)
